api/api/accounts/serializers.py

```diff
@@ -49,3 +49,84 @@ def create(self, validated_data):
             Staff.objects.create(user=user, **staff_data)
 
         return user
+
+    def update(self, instance, validated_data):
+        student_data = validated_data.pop('student', None)
+        faculty_data = validated_data.pop('faculty', None)
+        staff_data = validated_data.pop('staff', None)
+
+        self._update_base_user(instance, validated_data)
+
+        try:
+            student = instance.student
+        except Exception:
+            student = None
+
+        try:
+            faculty = instance.faculty
+        except Exception:
+            faculty = None
+
+        try:
+            staff = instance.staff
+        except Exception:
+            staff = None
+
+        if student:
+            self._update_student(student, student_data)
+        elif student_data:
+            Student.objects.create(user=instance, **student_data)
+        elif faculty:
+            self._update_faculty(faculty, faculty_data)
+        elif faculty_data:
+            Faculty.objects.create(user=instance, **faculty_data)
+        elif staff:
+            self._update_staff(staff, staff_data)
+        elif staff_data:
+            Staff.objects.create(user=instance, **staff_data)
+
+        return instance
+
+    def _update_base_user(self, instance, validated_data):
+        instance.email = validated_data.get('email', instance.email)
```

Comment:
`email`, `user_type` and `student.roll_no` should not be allowed to be changed, unless update is being done by an `admin` user. See if any other such field need to be considered.

Also, make sure that a user of one type does not add info about other type, like `user_type is student` and a `faculty` field is being updated.